resources/qml/SidebarHeader.qml

```diff
@@ -61,103 +61,123 @@ Column
-                        anchors.rightMargin: UM.Theme.getSize("default_margin").width / 2
-
-                        color: control.checked ? UM.Theme.getColor("tab_checked_text") :
-                                    control.pressed ? UM.Theme.getColor("tab_active_text") :
-                                    control.hovered ? UM.Theme.getColor("tab_hovered_text") : UM.Theme.getColor("tab_unchecked_text")
-
-                        font: UM.Theme.getFont("default")
-                        text: control.text
-                        elide: Text.ElideRight
+                        border.width: UM.Theme.getSize("default_lining").width
+                        border.color: control.checked ? UM.Theme.getColor("tab_checked_border") :
+                                            control.pressed ? UM.Theme.getColor("tab_active_border") :
+                                            control.hovered ? UM.Theme.getColor("tab_hovered_border") : UM.Theme.getColor("tab_unchecked_border")
+                        color: control.checked ? UM.Theme.getColor("tab_checked") :
+                                    control.pressed ? UM.Theme.getColor("tab_active") :
+                                    control.hovered ? UM.Theme.getColor("tab_hovered") : UM.Theme.getColor("tab_unchecked")
+                        Behavior on color { ColorAnimation { duration: 50; } }
+
+                        Rectangle
+                        {
+                            id: swatch
+                            visible: index > -1
+                            height: UM.Theme.getSize("setting_control").height / 2
+                            width: height
+                            anchors.left: parent.left
+                            anchors.leftMargin: (parent.height - height) / 2
+                            anchors.verticalCenter: parent.verticalCenter
+
+                            color: model.color
+                            border.width: UM.Theme.getSize("default_lining").width
+                            border.color: UM.Theme.getColor("setting_control_border")
+                        }
+
+                        Label
+                        {
+                            anchors.verticalCenter: parent.verticalCenter
+                            anchors.left: swatch.visible ? swatch.right : parent.left
+                            anchors.leftMargin: swatch.visible ? UM.Theme.getSize("default_margin").width / 2 : UM.Theme.getSize("default_margin").width
+                            anchors.right: parent.right
+                            anchors.rightMargin: UM.Theme.getSize("default_margin").width / 2
+
+                            color: control.checked ? UM.Theme.getColor("tab_checked_text") :
+                                        control.pressed ? UM.Theme.getColor("tab_active_text") :
+                                        control.hovered ? UM.Theme.getColor("tab_hovered_text") : UM.Theme.getColor("tab_unchecked_text")
```

Comment:
I see that this line has been copied over, but I still wouldn't mind if someone took the time to replace it with a clearer function.